aptos-move/aptos-vm-logging/src/lib.rs

```diff
@@ -55,6 +58,21 @@ impl SpeculativeEvent for VMLogEntry {
 static BUFFERED_LOG_EVENTS: Lazy<ArcSwapOption<SpeculativeEvents<VMLogEntry>>> =
     Lazy::new(|| ArcSwapOption::from(None));
 
+static DISABLE_SPECULATION: Lazy<AtomicBool> = Lazy::new(|| AtomicBool::new(false));
+
+/// Disables speculation, clears the BUFFERED_LOG_EVENTS and overrides the corresponding
+/// errors on accesses. Dispatches log events directly.
+/// This is useful to experiment with sharded block-stm, for example, as different shards
+/// interfere with each others' speculative logging. This seems like the least intrusive
+/// temporary workaround. TODO: more proper solution.
```

Comment:
@runtian-zhou comment: Be explicit that this SHOULD NOT BE USED (only one place that we add in this PR)